app/models/pxe_server.rb

```diff
@@ -34,7 +34,6 @@ def discovered_images
 
   def default_pxe_image_for_windows=(image)
     image.update(:default_for_windows => true)
-    clear_association_cache
```

Comment:
I don't see why this is here.  Do we need to reload `windows_images` and `pxe_images`?

Reply:
Yeah I think this can go away - I wrote a few specs to verify what exactly we're trying to do, and edge cases, and I think removing it is ok.  You ok if I push the specs to this PR?

Reply:
tl;dr, Since image is a parameter _technically_ it can come from anywhere, even outside of the cached collection.  However, since the `default_pxe_image_for_windows` getter does a find_by it always goes back to the database.  Having the specs in place will help if we change that later.

Reply:
Foe example, if we, in the future, change `default_pxe_image_for_windows`, then it will fail without some sort of cache clear.

```ruby
  def default_pxe_image_for_windows
    if pxe_images.loaded?
      pxe_images.detect(&:default_for_windows)
    else
      pxe_images.where(:default_for_windows => true).first
    end
  end
```

Reply:
Thanks...  yeah, can you push your specs to this branch.

Reply:
The fact we're passing a random image object, which can also be stale, updating it and then trying to clear associations which may or may not have that in memory image object complicates things.  Fetching from the database after methods can updates records seems like the safest path.

Reply:
Pushed the specs commit